resources/adoptopenjdk_install.rb

```diff
@@ -0,0 +1,202 @@
+#
+# Cookbook:: java
+# Resource:: adoptopenjdk_install
+#
+# Based on oracle_install.rb by Bryan W. Berry (<bryan.berry@gmail.com>)
+#
+
+resource_name :adoptopenjdk_install
+
+default_action :install
+
+property :url, String
+property :checksum, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
+property :md5, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
+property :app_home, String
+property :app_home_mode, Integer, default: 0755
+property :bin_cmds, Array, default: []
+property :owner, String, default: 'root'
+property :group, String, default: lazy { node['root_group'] }
+property :default, [true, false], default: true
+property :alternatives_priority, Integer, default: 1
+property :reset_alternatives, [true, false], default: true
+property :variant, ['hotspot', 'openj9', 'openj9-large-heap'], default: 'openj9'
+
+action :install do
+  raise 'No URL provided to download AdoptOpenJDK\'s tar file!' if new_resource.url.nil? || new_resource.url.empty?
+  app_dir_name, tarball_name, app_root, app_dir =
+    parse_dir_names(new_resource.url,
+                    new_resource.app_home,
+                    new_resource.variant)
+  app_group = new_resource.group
+  app_home = new_resource.app_home
+  Chef::Log.debug("processing #{new_resource.variant} variant")
+
+  directory app_root do
+    owner new_resource.owner
+    group app_group
+    mode new_resource.app_home_mode
+    recursive true
+    action :nothing
+  end.run_action(:create)
+
+  unless ::File.exist?(app_dir)
+    download_path = "#{Chef::Config[:file_cache_path]}/#{tarball_name}"
+    if adoptopendjk_downloaded?(download_path, new_resource)
+      Chef::Log.debug('AdoptOpenJDK tarball already downloaded, not downloading again')
+    else
+      Chef::Log.debug("downloading tarball from #{URI.parse(new_resource.url).host}")
+      remote_file "#{Chef::Config[:file_cache_path]}/#{tarball_name}" do
+        source new_resource.url
+        checksum new_resource.checksum
+        retries new_resource.retries
+        retry_delay new_resource.retry_delay
+        mode 0o644
+        action :nothing
+      end.run_action(:create_if_missing)
+    end
+
+    converge_by("extract compressed data into Chef file cache path and move extracted data to #{app_dir}") do
+      package 'tar' do
+        not_if { platform_family?('mac_os_x', 'windows') }
+        action :nothing
+      end.run_action(:install)
+
+      cmd = shell_out(%(tar xvzf "#{Chef::Config[:file_cache_path]}/#{tarball_name}" -C "#{Chef::Config[:file_cache_path]}" --no-same-owner)
```

Comment:
@tas50 I have been seeing more of this lately is this the prefered method? I am more familiar with the methods outlined [here](https://docs.chef.io/ruby.html) which might need an update or an expansion to help know when one is prefered over the other.